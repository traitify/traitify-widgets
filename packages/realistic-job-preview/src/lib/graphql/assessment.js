export const create = `
  mutation createAssessment(
    $localeKey: String!,
    $profileID: String!,
    $surveyID: String!
  ) {
    createAssessment(
      localeKey: $localeKey,
      profileId: $profileID,
      surveyId: $surveyID
    ) {
      completedAt
      failedReason
      id
      insertedAt
      instructions
      isFit
      jobId
      localeKey
      profileId
      responses {
        id
        questionId
        questionText
        responseOptions {
          responseOptionId
          responseOptionText
        }
        selectedResponseOptionId
        selectedResponseOptionText
      }
      startedAt
      status
      surveyId
      totalCorrectResponses
      updatedAt
      videos {
        thumbnailUrl
        videoUrl
      }
    }
  }
`;

export const get = `
  query assessment(
    $assessmentID: ID!
  ) {
    assessment(
      id: $assessmentID
    ) {
      completedAt
      failedReason
      id
      insertedAt
      instructions
      isFit
      jobId
      localeKey
      profileId
      responses {
        id
        questionId
        questionText
        responseOptions {
          responseOptionId
          responseOptionText
        }
        selectedResponseOptionId
        selectedResponseOptionText
      }
      startedAt
      status
      surveyId
      totalCorrectResponses
      updatedAt
      videos {
        thumbnailUrl
        videoUrl
      }
    }
  }
`;

export const path = "/realistic-job-previews/graphql";

export const start = `
  mutation startAssessment(
    $assessmentID: ID!
  ) {
    startAssessment(
      id: $assessmentID
    ) {
      completedAt
      failedReason
      id
      insertedAt
      instructions
      isFit
      jobId
      localeKey
      profileId
      responses {
        id
        questionId
        questionText
        responseOptions {
          responseOptionId
          responseOptionText
        }
        selectedResponseOptionId
        selectedResponseOptionText
      }
      startedAt
      status
      surveyId
      totalCorrectResponses
      updatedAt
      videos {
        thumbnailUrl
        videoUrl
      }
    }
  }
`;

export const update = `
  mutation updateAssessmentAnswer(
    $answers: [AssessmentAnswer],
    $assessmentID: String!
  ) {
    updateAssessmentAnswer(
      answers: $answers,
      assessmentId: $assessmentID
    ) {
      completedAt
      failedReason
      id
      insertedAt
      instructions
      isFit
      jobId
      localeKey
      profileId
      responses {
        id
        questionId
        questionText
        responseOptions {
          responseOptionId
          responseOptionText
        }
        selectedResponseOptionId
        selectedResponseOptionText
      }
      startedAt
      status
      surveyId
      totalCorrectResponses
      updatedAt
      videos {
        thumbnailUrl
        videoUrl
      }
    }
  }
`;
