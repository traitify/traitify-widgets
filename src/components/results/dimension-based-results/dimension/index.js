import PropTypes from "prop-types";
import {Component} from "react";
import {detailWithPerspective} from "lib/helpers";
import TraitifyPropTypes from "lib/helpers/prop-types";
import withTraitify from "lib/with-traitify";
import {rgba} from "lib/helpers/color";
import style from "./style.scss";

class Dimension extends Component {
  static defaultProps = {assessmentID: null};
  static propTypes = {
    assessmentID: PropTypes.string,
    getOption: PropTypes.func.isRequired,
    index: PropTypes.number.isRequired,
    translate: PropTypes.func.isRequired,
    type: PropTypes.shape({
      personality_type: PropTypes.shape({
        badge: PropTypes.shape({
          color_1: PropTypes.string.isRequired,
          image_medium: PropTypes.string.isRequired
        }).isRequired,
        details: PropTypes.arrayOf(
          PropTypes.shape({
            body: PropTypes.string.isRequired,
            title: PropTypes.string.isRequired
          })
        ).isRequired,
        level: PropTypes.string.isRequired,
        name: PropTypes.string.isRequired
      }).isRequired,
      score: PropTypes.number.isRequired
    }).isRequired,
    ui: TraitifyPropTypes.ui.isRequired
  };
  constructor(props) {
    super(props);

    this.state = {showContent: props.index === 0};
  }
  componentDidMount() {
    this.props.ui.trigger("Dimension.initialized", this);
  }
  componentDidUpdate(prevProps) {
    this.props.ui.trigger("Dimension.updated", this);

    if(this.props.assessmentID !== prevProps.assessmentID) {
      this.setState({showContent: this.props.index === 0});
    }
  }
  trigger = () => {
    this.props.ui.trigger("Dimension.showContent", this, this.props.type.personality_type);
    this.setState((state) => ({showContent: !state.showContent}));
  };
  render() {
    const type = this.props.type.personality_type;
    const color = `#${type.badge.color_1}`;
    const options = {base: type, perspective: this.props.getOption("perspective")};
    const description = detailWithPerspective({...options, name: "description"});
    const shortDescription = detailWithPerspective({...options, name: "short_description"});
    const benefits = [];
    const pitfalls = [];

    type.details.forEach((detail) => {
      if(detail.title === "Benefits") { benefits.push(detail.body); }
      if(detail.title === "Pitfalls") { pitfalls.push(detail.body); }
    });

    return (
      <li className={style.dimension}>
        <div
          className={style.main}
          style={{background: rgba(color, 8.5), borderLeft: `5px solid ${color}`}}
        >
          <div className={style.side}>
            <p className={style.icon}>
              <img src={type.badge.image_medium} alt="" ariahidden="true" />
            </p>
          </div>
          <div className={style.content}>
            <h2 className={style.title}>
              {type.name} <span style={{color}}>|</span> {this.props.type.score} - {type.level}
            </h2>
            <p className={style.description}>{shortDescription}</p>
            <p className={style.triggerButton}>
              <button
                className={style.trigger}
                style={{background: rgba(color, 30)}}
                onClick={this.trigger}
                type="button"
              >
                {this.props.translate(this.state.showContent ? "show_less" : "show_more")}
              </button>
            </p>
          </div>
        </div>
        {this.state.showContent && (
          <div className={style.details}>
            <div className={style.content} style={{background: rgba(color, 30)}}>
              <div className={style.extendedDesc}>
                <h3>{this.props.translate("extended_description")}</h3>
                <p className={style.description}>{description}</p>
              </div>
              <div className={style.detail}>
                <h4 className={style.benefits}>{this.props.translate("potential_benefits")}</h4>
                <ul>
                  {benefits.map((benefit) => (<li key={benefit}>{benefit}</li>))}
                </ul>
              </div>
              <div className={style.detail}>
                <h4 className={style.pitfalls}>{this.props.translate("potential_pitfalls")}</h4>
                <ul>
                  {pitfalls.map((pitfall) => (<li key={pitfall}>{pitfall}</li>))}
                </ul>
              </div>
            </div>
          </div>
        )}
      </li>
    );
  }
}

export {Dimension as Component};
export default withTraitify(Dimension);
