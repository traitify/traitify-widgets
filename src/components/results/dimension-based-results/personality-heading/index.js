import PropTypes from "prop-types";
import {Component} from "react";
import DangerousHTML from "lib/helpers/dangerous-html";
import TraitifyPropTypes from "lib/helpers/prop-types";
import withTraitify from "lib/with-traitify";
import style from "./style.scss";

class PersonalityHeading extends Component {
  static defaultProps = {assessment: null, deck: null};
  static propTypes = {
    assessment: PropTypes.shape({
      archetype: PropTypes.shape({
        name: PropTypes.string.isRequired
      })
    }),
    deck: PropTypes.shape({name: PropTypes.string.isRequired}),
    followDeck: PropTypes.func.isRequired,
    isReady: PropTypes.func.isRequired,
    translate: PropTypes.func.isRequired,
    ui: TraitifyPropTypes.ui.isRequired
  };
  componentDidMount() {
    this.props.ui.trigger("PersonalityHeading.initialized", this);
    this.props.followDeck();
  }
  componentDidUpdate() {
    this.props.ui.trigger("PersonalityHeading.updated", this);
  }
  render() {
    if(!this.props.isReady("results")) { return null; }
    if(!this.props.isReady("deck")) { return null; }

    const personality = this.props.assessment.archetype;
    if(!personality) { return null; }

    return (
      <div className={style.personality}>
        <DangerousHTML
          className={style.content}
          html={this.props.translate("personality_heading", {
            deck_name: this.props.deck.name,
            personality: `<strong>${personality.name}</strong>`
          })}
        />
      </div>
    );
  }
}

export {PersonalityHeading as Component};
export default withTraitify(PersonalityHeading);
