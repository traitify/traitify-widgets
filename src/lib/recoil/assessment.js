import {selector, selectorFamily} from "recoil";
import {assessmentsState} from "./assessments";
import {
  activeState,
  activeIDState,
  activeTypeState,
  cacheState,
  graphqlState,
  httpState,
  localeState,
  safeCacheKeyState
} from "./base";

// TODO: Return error instead of null for cognitive and external
// return {errors: response.errors, id};
export const cognitiveAssessmentQuery = selectorFamily({
  get: (id) => async({get}) => {
    if(!id) { return null; }

    const cache = get(cacheState);
    const cacheKey = get(safeCacheKeyState({id, type: "assessment"}));
    const cached = cache.get(cacheKey);
    if(cached) { return cached; }

    const GraphQL = get(graphqlState);
    const http = get(httpState);
    const params = {
      query: GraphQL.cognitive.get,
      variables: {localeKey: get(localeState), testID: id}
    };
    const response = await http.post({path: GraphQL.cognitive.path, params});
    if(response.errors) {
      console.warn("cognitive-assessment", response.errors); /* eslint-disable-line no-console */
      return null;
    }

    const assessment = response.data.cognitiveTest;
    if(!assessment?.completed) { return assessment; }

    cache.set(cacheKey, assessment);

    return assessment;
  },
  key: "assessment/cognitive"
});

export const externalAssessmentQuery = selectorFamily({
  get: (id) => async({get}) => {
    if(!id) { return null; }

    const cache = get(cacheState);
    const cacheKey = get(safeCacheKeyState({id, type: "assessment"}));
    const cached = cache.get(cacheKey);
    if(cached) { return cached; }

    const GraphQL = get(graphqlState);
    const http = get(httpState);
    const query = {
      params: {
        query: GraphQL.external.get,
        variables: {id}
      },
      path: GraphQL.external.path
    };
    if(http.version === "v1") { query.version = GraphQL.external.version; }

    const response = await http.post(query);
    if(response.errors) {
      console.warn("external-assessment", response.errors); /* eslint-disable-line no-console */
      return null;
    }

    const assessment = response.data.getAssessment;
    if(!assessment?.completedAt) { return assessment; }

    cache.set(cacheKey, assessment);

    return assessment;
  },
  key: "assessment/external"
});

export const personalityAssessmentQuery = selectorFamily({
  get: (id) => async({get}) => {
    if(!id) { return null; }

    const cache = get(cacheState);
    const cacheKey = get(safeCacheKeyState({id, type: "assessment"}));
    const cached = cache.get(cacheKey);
    if(cached) { return cached; }

    const params = {
      data: "archetype,blend,instructions,recommendation,slides,types,traits",
      locale_key: get(localeState)
    };
    const http = get(httpState);
    const response = await http.get({path: `/assessments/${id}`, params});
    if(!response?.completed_at) { return response; }

    cache.set(cacheKey, response);

    return response;
  },
  key: "assessment/personality"
});

export const genericAssessmentQuery = selectorFamily({
  get: (id) => async({get}) => {
    if(!id) { return null; }

    const cache = get(cacheState);
    const cacheKey = get(safeCacheKeyState({id, type: "assessment"}));
    const cached = cache.get(cacheKey);
    if(cached && !cached.completed) { return cached; }

    const GraphQL = get(graphqlState);
    const http = get(httpState);
    const params = {
      query: GraphQL.generic.questions,
      variables: {assessmentID: id}
    };

    const response = await http.post({path: GraphQL.generic.path, params});
    if(response.errors) {
      console.warn("generic-assessment", response.errors); /* eslint-disable-line no-console */
      return null;
    }

    const assessment = response.data.genericSurveyQuestions;
    if(!assessment?.completedAt) { return assessment; }

    const query = GraphQL.generic.result;
    const variables = {assessmentID: id};

    const resultResponse = await http.post(GraphQL.generic.path, {query, variables});
    if(resultResponse.errors) { return null; }
    const result = resultResponse.data.genericAssessmentResult.assessment;

    cache.set(cacheKey, result);

    return result;
  },
  key: "assessment/generic"
});

export const assessmentQuery = selectorFamily({
  get: ({id, surveyType}) => async({get}) => {
    if(surveyType === "cognitive") { return get(cognitiveAssessmentQuery(id)); }
    if(surveyType === "external") { return get(externalAssessmentQuery(id)); }
    if(surveyType === "generic") { return get(genericAssessmentQuery(id)); }
    if(surveyType === "personality") { return get(personalityAssessmentQuery(id)); }

    return null;
  },
  key: "assessment"
});

export const activeAssessmentQuery = selector({
  get: async({get}) => {
    const id = get(activeIDState);
    const surveyType = get(activeTypeState);

    return get(assessmentQuery({id, surveyType}));
  },
  key: "assessment/active"
});

export const completedAssessmentQuery = selectorFamily({
  get: ({surveyType}) => async({get}) => {
    if(!surveyType) {
      const active = get(activeState);
      if(!active?.completed) { return null; }

      return get(activeAssessmentQuery);
    }

    const assessments = get(assessmentsState);
    if(!assessments) { return null; }

    const assessment = assessments
      .find(({completed, surveyType: type}) => type === surveyType && completed);
    if(!assessment) { return null; }

    return get(assessmentQuery({id: assessment.id, surveyType}));
  },
  key: "assessment/completed"
});
