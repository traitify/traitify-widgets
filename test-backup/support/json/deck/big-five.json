{
  "assessment_type": "DIMENSION_BASED",
  "scoring_scale": "STEN",
  "id": "big-five",
  "name": "Big Five",
  "description": "Traitify's Big Five Personality Assessment provides a powerful profile of an individual's work-related behavior. It measures Openness, Conscientiousness, Extraversion, Agreeableness and Emotional Stability. The Big Five is backed by 50 years of research and align with key job and performance metrics. The questionnaire allows users to benchmark job candidates against industry standard profiles and/or a businesses' top performers, to focus interview questions and much more. Within organizations it has applications in team building, exploring communication styles, and pinpointing areas for personal development.  It is an essential tool for HR professionals, L&D specialists, and anyone who wants to know more about work personality.",
  "personality_group": "Big Five",
  "slide_count": 73,
  "badges": [
    {
      "image_small": "https://cdn.traitify.com/badges/c3e5148a-da49-4d3c-abd7-94dadf9ece2b_small",
      "image_medium": "https://cdn.traitify.com/badges/96dbf653-6ea0-4dc0-94dc-f6db3247510c_medium",
      "image_large": "https://cdn.traitify.com/badges/b87d4a86-198d-4eb0-b5c9-99f2dc346ef9_large",
      "font_color": "",
      "color_1": "29B770",
      "color_2": "",
      "color_3": "",
      "personality_type": "Agreeableness",
      "description": "Influencing people (Agreeableness) concerns the way someone balances their emotional understanding of other people, and their respect for differing viewpoints, with the style in which they try to influence or negotiate with them."
    },
    {
      "image_small": "https://cdn.traitify.com/badges/0902c419-92a0-4855-8ff2-2e9875046556_small",
      "image_medium": "https://cdn.traitify.com/badges/0d29bdb5-b502-4036-a790-9479d588e8fb_medium",
      "image_large": "https://cdn.traitify.com/badges/1488d081-2d84-42aa-b7ee-d58facba08b5_large",
      "font_color": "",
      "color_1": "F98B1C",
      "color_2": "",
      "color_3": "",
      "personality_type": "Conscientiousness",
      "description": "Delivering Results (Conscientiousness) concerns work style and how someone directs their efforts and attention toward completing tasks. The level of discipline, dedication, perseverance, organization and reliability are hallmark factors of this dimension. "
    },
    {
      "image_small": "https://cdn.traitify.com/badges/2f793151-b520-4cc9-91c8-5ddb0c091c94_small",
      "image_medium": "https://cdn.traitify.com/badges/16cd30cc-3d5c-4c7e-b13f-828853034334_medium",
      "image_large": "https://cdn.traitify.com/badges/95f6ff51-4de1-4146-9980-eea84d276f7a_large",
      "font_color": "",
      "color_1": "ef615e",
      "color_2": "",
      "color_3": "",
      "personality_type": "Emotional Stability",
      "description": "Managing pressure (Emotional Stability) concerns the manner in which someone deals with pressure and the way in which they control their emotions and underlying tension, in order to stay on task and cope with everyday challenges."
    },
    {
      "image_small": "https://cdn.traitify.com/badges/5e021131-c2d9-4190-a7bb-9277d5a159ec_small",
      "image_medium": "https://cdn.traitify.com/badges/bf796c03-a863-4f61-9a1a-1f4fe6e408e8_medium",
      "image_large": "https://cdn.traitify.com/badges/d7f1a092-1d74-4b24-8777-3cff52457c7a_large",
      "font_color": "",
      "color_1": "FFCC3B",
      "color_2": "",
      "color_3": "",
      "personality_type": "Extraversion",
      "description": "Engaging with people (Extraversion) concerns someoneâ€™s interest, investment and comfort in developing relationships with others - customers, clients, work groups or colleagues."
    },
    {
      "image_small": "https://cdn.traitify.com/badges/c05e769f-2c6d-42e3-83b7-6ca4239e3da0_small",
      "image_medium": "https://cdn.traitify.com/badges/216e23f2-f27c-46ae-b4b1-00c97a01db22_medium",
      "image_large": "https://cdn.traitify.com/badges/89aab732-f414-4f08-ae83-e878580f0ed2_large",
      "font_color": "",
      "color_1": "42B0DB",
      "color_2": "",
      "color_3": "",
      "personality_type": "Openness",
      "description": "Solving Problems (Openness) concerns how someone thinks about work problems, projects and challenges, how receptive they are to new or different information or approaches, and the way this influences their decision-making."
    }
  ],
  "example_usage": null,
  "validated": false,
  "image_cover": "https://cdn.traitify.com/decks/fd6f1744-bdc7-48c0-ab48-897ac542d2f8_cover",
  "personality_traits": [],
  "personality_types": []
}
