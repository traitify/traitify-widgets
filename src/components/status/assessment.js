import {faCheck} from "@fortawesome/free-solid-svg-icons";
import PropTypes from "prop-types";
import {useSetRecoilState} from "recoil";
import Icon from "components/common/icon";
import get from "lib/common/object/get";
import useListener from "lib/hooks/use-listener";
import useOption from "lib/hooks/use-option";
import {activeState} from "lib/recoil";
import style from "./style.scss";

// TODO: Extract text to translate
const translations = {
  complete: "Complete",
  loading: "Loading", // Already translated
  status: {
    start: "Start Assessment"
  },
  survey: {
    cognitive_assessment: "Cognitive Assessment",
    external_assessment: "External Assessment",
    personality_assessment: "Personality Assessment"
  }
};

function Button({assessment}) {
  const listener = useListener();
  const options = useOption("status") || {};
  const redirect = get(options, "allowRedirect", true);
  const setActive = useSetRecoilState(activeState);

  if(assessment.completed) {
    return <button disabled={true} type="button">{translations.complete}</button>;
  }

  if(assessment.skipped) {
    return <button disabled={true} type="button">Skipped</button>;
  }

  if(assessment.link && redirect) {
    return <a href={assessment.link}>{translations.status.start}</a>;
  }

  if(assessment.loading) {
    return <button disabled={true} type="button">{translations.loading}</button>;
  }

  const start = () => {
    listener.trigger("Survey.start", {assessment});
    setActive({...assessment});
  };

  return <button onClick={start} type="button">{translations.status.start}</button>;
}

Button.propTypes = {
  assessment: PropTypes.shape({
    completed: PropTypes.bool,
    link: PropTypes.string,
    loading: PropTypes.bool,
    skipped: PropTypes.bool
  }).isRequired
};

function Assessment({assessment}) {
  const surveyName = assessment.surveyName || translations.survey[`${assessment.surveyType}_assessment`];

  return (
    <div className={[style.assessment, assessment.completed && style.inactive].filter(Boolean).join(" ")}>
      <div>
        {assessment.completed && <Icon className={style.icon} icon={faCheck} />}
        <span className={style.text}>{surveyName}</span>
      </div>
      <Button assessment={assessment} />
    </div>
  );
}

Assessment.propTypes = {
  assessment: PropTypes.shape({
    completed: PropTypes.bool,
    skipped: PropTypes.bool,
    surveyName: PropTypes.string,
    surveyType: PropTypes.string.isRequired
  }).isRequired
};

export default Assessment;
