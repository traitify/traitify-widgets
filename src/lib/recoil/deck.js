import {noWait, selector} from "recoil";
import {assessmentQuery} from "./assessment";
import {
  activeTypeState,
  cacheState,
  httpState,
  localeState,
  safeCacheKeyState
} from "./base";

export const deckIDState = selector({
  get: ({get}) => {
    const type = get(activeTypeState);
    if(type !== "personality") { return null; }

    const loadable = get(noWait(assessmentQuery));
    if(loadable.state !== "hasValue") { return null; }

    return loadable.contents?.deck_id;
  },
  key: "deck-id"
});

export const deckQuery = selector({
  get: async({get}) => {
    const deckID = await get(deckIDState);
    if(!deckID) { return null; }

    const cache = get(cacheState);
    const cacheKey = get(safeCacheKeyState({id: deckID, type: "deck"}));
    const cached = cache.get(cacheKey);
    if(cached) { return cached; }

    const params = {locale_key: get(localeState)};
    const http = get(httpState);
    const response = await http.get(`/decks/${deckID}`, params);
    if(response) { cache.set(cacheKey, response); }

    return response;
  },
  key: "deck"
});
