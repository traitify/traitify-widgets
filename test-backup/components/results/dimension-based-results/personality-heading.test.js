import {Component} from "components/results/dimension-based-results/personality-heading";
import ComponentHandler from "support/component-handler";
import assessment from "support/json/assessment/dimension-based.json";
import deck from "support/json/deck/big-five.json";

jest.mock("lib/with-traitify", () => ((value) => value));

describe("PersonalityHeading", () => {
  let props;

  beforeEach(() => {
    props = {
      assessment,
      deck,
      followDeck: jest.fn().mockName("followDeck"),
      isReady: jest.fn().mockName("isReady").mockImplementation(() => true),
      translate: jest.fn().mockName("translate").mockImplementation((value, options = {}) => `${value}, ${options}`),
      ui: {
        current: {},
        off: jest.fn().mockName("off"),
        on: jest.fn().mockName("on"),
        trigger: jest.fn().mockName("trigger")
      }
    };
  });

  describe("callbacks", () => {
    it("triggers initialization", () => {
      const component = new ComponentHandler(<Component {...props} />);

      expect(props.ui.trigger).toHaveBeenCalledWith("PersonalityHeading.initialized", component.instance);
    });

    it("triggers update", () => {
      const component = new ComponentHandler(<Component {...props} />);
      component.updateProps();

      expect(props.ui.trigger).toHaveBeenCalledWith("PersonalityHeading.updated", component.instance);
    });
  });

  it("follows the deck", () => {
    new ComponentHandler(<Component {...props} />);

    expect(props.followDeck).toHaveBeenCalled();
  });

  it("renders component", () => {
    const component = new ComponentHandler(<Component {...props} />);

    expect(component.tree).toMatchSnapshot();
  });

  it("renders nothing if deck not ready", () => {
    props.deck = null;
    props.isReady.mockImplementation((value) => value !== "deck");
    const component = new ComponentHandler(<Component {...props} />);

    expect(component.tree).toMatchSnapshot();
  });

  it("renders nothing if results not ready", () => {
    props.assessment = null;
    props.isReady.mockImplementation((value) => value !== "results");
    const component = new ComponentHandler(<Component {...props} />);

    expect(component.tree).toMatchSnapshot();
  });

  it("renders nothing if no archetype", () => {
    props.assessment = {...props.assessment, archetype: null};
    const component = new ComponentHandler(<Component {...props} />);

    expect(component.tree).toMatchSnapshot();
  });
});
