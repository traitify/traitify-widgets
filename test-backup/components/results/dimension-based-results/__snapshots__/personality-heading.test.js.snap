// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PersonalityHeading renders component 1`] = `
<div
  className="personality"
>
  <div
    className="content"
    dangerouslySetInnerHTML={
      {
        "__html": "personality_heading, [object Object]",
      }
    }
  />
</div>
`;

exports[`PersonalityHeading renders nothing if deck not ready 1`] = `null`;

exports[`PersonalityHeading renders nothing if no archetype 1`] = `null`;

exports[`PersonalityHeading renders nothing if results not ready 1`] = `null`;
